diff --git a/node_modules/dsp.js/dsp.js b/node_modules/dsp.js/dsp.js
index 7188235..c95072d 100644
--- a/node_modules/dsp.js/dsp.js
+++ b/node_modules/dsp.js/dsp.js
@@ -57,13 +57,13 @@ var DSP = {
 function setupTypedArray(name, fallback) {
   // check if TypedArray exists
   // typeof on Minefield and Chrome return function, typeof on Webkit returns object.
-  if (typeof this[name] !== "function" && typeof this[name] !== "object") {
+  if (typeof globalThis[name] !== "function" && typeof globalThis[name] !== "object") {
     // nope.. check if WebGLArray exists
-    if (typeof this[fallback] === "function" && typeof this[fallback] !== "object") {
-      this[name] = this[fallback];
+    if (typeof globalThis[fallback] === "function" && typeof globalThis[fallback] !== "object") {
+      globalThis[name] = globalThis[fallback];
     } else {
       // nope.. set as Native JS array
-      this[name] = function(obj) {
+      globalThis[name] = function(obj) {
         if (obj instanceof Array) {
           return obj;
         } else if (typeof obj === "number") {
